src/core/lib/config/core_configuration.h

```diff
@@ -19,6 +19,7 @@
 
 #include <atomic>
 
+#include "src/core/ext/xds/xds_channel_creds.h"
```

Comment:
We should not have code from src/core/lib depending on code in src/core/ext.  Instead, I suggest moving xds_channel_creds_registry.{cc,h} and xds_channel_default_channel_creds.cc from src/core/ext/xds to src/core/lib/security/credentials.  You can drop the "xds_" prefix from the filenames and the "Xds" prefix from the class names, since this mechanism isn't necessarily xDS-specific.

Reply:
Done, I moved them to src/core/lib/security/credentials/xds with no change, PTAL.

Reply:
No, none of that code is xDS-specific.  Please remove the `Xds` or `xds_` prefixes from all types, functions, variables, and filenames in src/core/lib.

Reply:
Sure, updated. I still left calls to RegisterChannelDefaultCreds() in grpc_plugin_registry.cc inside the GRPC_NO_XDS macro since it's only required with xDS, let me know if you want to move them out.